### PR TITLE
CDAP-1069: Document a limitation in using Datasets in Spark

### PR DESCRIPTION
**Pull request by gokulavasan**
https://issues.cask.co/browse/CDAP-1069

@awholegunch Does this limitation/known issue be documented here? Or should it be documented somewhere else?

@anew Can you please review the content?

### CODE REVIEW

File: cdap-docs/developers-manual/source/building-blocks/spark-programs.rst [hunk 1]
```diff
@@ -98,6 +98,10 @@ Spark programs in CDAP can directly access **Dataset** similar to the way a MapR
 Procedure can. These programs can create Spark's Resilient Distributed Dataset (RDD) by
 reading a Dataset and can also write RDD to a Dataset.
 
+In order to access a Dataset in Spark, both the key and value classes has to be serializable.
+Otherwise, Spark will fail to read or write them.
+For example, in Table Dataset, the value type is Row which is not Serializable.
```
Comment (anew) [inline cdap-docs/developers-manual/source/building-blocks/spark-programs.rst]: , the Table Dataset has a value type of Row...

**Comment by anew** [comment]
content's correct, but I have a comment on grammar.

File: cdap-docs/developers-manual/source/building-blocks/spark-programs.rst [hunk 2]
```diff
@@ -98,6 +98,10 @@ Spark programs in CDAP can directly access **Dataset** similar to the way a MapR
 Procedure can. These programs can create Spark's Resilient Distributed Dataset (RDD) by
 reading a Dataset and can also write RDD to a Dataset.
 
+In order to access a Dataset in Spark, both the key and value classes has to be serializable.
```
Comment (anew) [inline cdap-docs/developers-manual/source/building-blocks/spark-programs.rst]: sorry, overlooked this first time. It is plural so it has to be ...have to be serializable.

File: cdap-docs/developers-manual/source/building-blocks/spark-programs.rst [hunk 3]
```diff
@@ -98,6 +98,10 @@ Spark programs in CDAP can directly access **Dataset** similar to the way a MapR
 Procedure can. These programs can create Spark's Resilient Distributed Dataset (RDD) by
 reading a Dataset and can also write RDD to a Dataset.
 
+In order to access a Dataset in Spark, both the key and value classes have to be serializable.
+Otherwise, Spark will fail to read or write them.
+For example, the Table Dataset has a value type of Row, which is not serializable.
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/spark-programs.rst]:
````
For example, the ``Table`` Dataset has a value type of ``Row``, which is not serializable. An ``ObjectStore`` dataset can be used, as its classes are serializable.
````

**Comment by awholegunch** [comment]
@anew Can you check my change?

**Comment by anew** [comment]
Are you sure about the ObjectStore? Have you tested that?

**Comment by anew** [comment]
As discussed, an ObjectStore can only be used if the objects stored in it are serializable.

File: cdap-docs/developers-manual/source/building-blocks/spark-programs.rst [hunk 4]
```diff
@@ -98,6 +98,11 @@ Spark programs in CDAP can directly access **Dataset** similar to the way a MapR
...
+An ``ObjectStore`` dataset could be used, provided its classes are serializable.
```
Comment (anew) [inline cdap-docs/developers-manual/source/building-blocks/spark-programs.rst]: could? or can?

**Comment by anew** [comment]
LGTM

**Pull request merged**